pkg/extension/opampextension/opamp_agent_test.go

```diff
@@ -159,435 +49,482 @@ func TestApplyRemoteConfig(t *testing.T) {
 		},
 		ConfigHash: []byte("b2b1e3e7f45d564db1c0b621bbf67008"),
 	}
-
-	changed, err := o.applyRemoteConfig(rc)
-	assert.NoError(t, err)
-	assert.True(t, changed)
-	assert.NotEqual(t, len(o.effectiveConfig), 0)
-
-	cfg.AcceptsRemoteConfiguration = false
-	changed, err = o.applyRemoteConfig(rc)
-	assert.False(t, changed)
-	assert.Error(t, err)
-	assert.Equal(t, "OpAMP agent does not accept remote configuration", err.Error())
+	return rc
 }
 
-func TestApplyRemoteApacheConfig(t *testing.T) {
-	d, err := os.MkdirTemp("", "opamp.d")
-	assert.NoError(t, err)
-	defer os.RemoveAll(d)
-
-	cfg := createDefaultConfig().(*Config)
-	cfg.RemoteConfigurationDirectory = d
-	set := extensiontest.NewNopSettings()
-	o, err := newOpampAgent(cfg, set.Logger, set.BuildInfo, set.Resource)
-	assert.NoError(t, err)
-
-	assert.Equal(t, len(o.effectiveConfig), 0)
-
-	path := filepath.Join("testdata", "opamp.d", "opamp-apache-config.yaml")
-	rb, err := os.ReadFile(path)
-	assert.NoError(t, err)
-
-	rc := &protobufs.AgentRemoteConfig{
-		Config: &protobufs.AgentConfigMap{
-			ConfigMap: map[string]*protobufs.AgentConfigFile{
-				"default": {
-					Body: rb,
-				},
+func TestOpampAgent(t *testing.T) {
+	testCases := []struct {
+		name     string
+		setup    func() (*Config, extension.Settings)
+		validate func(*opampAgent, *testing.T)
+	}{
+		{
+			name: "NewOpampAgent",
+			setup: func() (*Config, extension.Settings) {
+				cfg := createDefaultConfig()
+				set := extensiontest.NewNopSettings()
+				set.BuildInfo = component.BuildInfo{Version: "test version", Command: "otelcoltest"}
+				return cfg.(*Config), set
+			},
```

Comment:
I see the setup is the same or almost the same across many tests? Can we unify it, so we put it into test body and not as test argument?

If there are more groups (2) we can split them to two tests

Reply:
addressed